### ISSUE TITLE
WHOLE-INDEX false negative with CAN-FIND

### ISSUE THREAD

**Issue by movedoa** [Riverside-Software/sonar-openedge#768]
In the example code below, only the third CAN-FIND triggers the WHOLE-INDEX rule.
This also happens with Database buffers so it has nothing to do with using a temp-table.

```
CLASS WholeIndexBug:

   DEFINE PRIVATE TEMP-TABLE ttFoo NO-UNDO
      FIELD Field1 AS CHARACTER
      FIELD Field2 AS CHARACTER
   .

   METHOD PUBLIC LOGICAL HasVisibleMaskItems(
   ):
      /* NO WHOLE-INDEX */
      CAN-FIND(
         FIRST ttFoo WHERE
           ttFoo.Field1 = "Foo":U
         NO-LOCK
      ).

      /* NO WHOLE-INDEX */
      DEFINE VARIABLE llFoo AS LOGICAL NO-UNDO.
      llFoo = CAN-FIND(
         FIRST ttFoo WHERE
           ttFoo.Field1 = "Foo":U
         NO-LOCK
      ).

      /* WHOLE-INDEX! */
      RETURN CAN-FIND(
         FIRST ttFoo WHERE
            ttFoo.Field1 = "Foo":U
         NO-LOCK
      ).
   END METHOD.

END CLASS.
```

**Comment by gquerret** [comment]
Reproduced and confirmed, will be fixed quickly.
Line numbers in XREF are reported on the line number of the enclosing statement, and the first two statements are "undeclared" statements (expression without assignment, and assignment without the assign keyword).

**Comment by gquerret** [comment]
Merged in develop
